### PR TITLE
syslog: add syslog channel filtering function

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
support to control the opening or closing of the specified channel through the syslogmask command at runtime

### CODE REVIEW

**Comment by acassis** [comment]
@Gary-Hobson I think you missed to include a Documentation here: https://nuttx.apache.org/docs/latest/components/drivers/special/syslog.html

File: drivers/syslog/syslog_channel.c [hunk 1]
```diff
@@ -272,6 +288,14 @@ int syslog_channel(FAR struct syslog_channel_s *channel)
         {
           if (g_syslog_channel[i] == NULL)
             {
+#  ifdef CONFIG_SYSLOG_IOCTL
+              if (channel->sc_name[0] == '\0')
+                {
+                  snprintf(channel->sc_name, sizeof(channel->sc_name),
+                          "channel-%p", channel->sc_ops);
```
Comment (pkarashchenko) [inline drivers/syslog/syslog_channel.c:295]:
```suggestion
                  snprintf(channel->sc_name, sizeof(channel->sc_name),
                           "channel-%p", channel->sc_ops);
```

File: drivers/syslog/syslog_chardev.c [hunk 2]
```diff
@@ -71,6 +80,62 @@ static ssize_t syslog_chardev_write(FAR struct file *filep,
   return len;
 }
 
+#ifdef CONFIG_SYSLOG_IOCTL
+static int syslog_chardev_ioctl(FAR struct file *filep,
+                                int cmd, unsigned long arg)
+{
+  FAR struct syslog_channel_info_s *info;
+  struct syslog_channel_s *channel = NULL;
```
Comment (pkarashchenko) [inline drivers/syslog/syslog_chardev.c:88]:
```suggestion
  FAR struct syslog_channel_s *channel = NULL;
```

File: drivers/syslog/syslog_channel.c (hunk 1 above)
Comment (Gary-Hobson) [reply drivers/syslog/syslog_channel.c]: Done

**Review by acassis — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**